dhis-2/dhis-api/src/main/java/org/hisp/dhis/translation/TranslationProperty.java

```diff
@@ -33,14 +33,19 @@
  */
 public enum TranslationProperty
 {
-    NAME ( "name" ),
-    SHORT_NAME ( "shortName" ),
-    DESCRIPTION ("description" ),
-    FORM_NAME ( "formName" ),
-    NUMERATOR_DESCRIPTION ( "numeratorDescription" ),
-    DENOMINATOR_DESCRIPTION ( "denominatorDescription" ),
-    RELATIONSHIP_FROM_TO_NAME ("fromToName" ),
-    RELATIONSHIP_TO_FROM_NAME ("toFromName" );
+    NAME( "name" ),
+    SHORT_NAME( "shortName" ),
+    DESCRIPTION( "description" ),
+    FORM_NAME( "formName" ),
+    NUMERATOR_DESCRIPTION( "numeratorDescription" ),
+    DENOMINATOR_DESCRIPTION( "denominatorDescription" ),
+    RELATIONSHIP_FROM_TO_NAME( "fromToName" ),
+    RELATIONSHIP_TO_FROM_NAME( "toFromName" ),
+    INSTRUCTION( "instruction" ),
+    PROGRAM_INCIDENT_DATE_LABEL( "programIncidentDateLabel" ),
```

Comment:
So.. The current translation solution has a weakness in that if two entities have properties with the same name, and a translation key exists, it could be applied to both entities, even if just one of the entities are supposed to be translated.

Still we have not prefixed these properties with the entity name, e.g. `FORM_NAME` is just for data element but we have not prefixed the property with DATA_ELEMENT.

By having prefixed names like `PROGRAM_INCIDENT_DATE_LABEL` we are breaking that convention. Even if not perfect, I am thinking we should stick to the convention and use `INCIDENT_DATE_LABEL` and so on.